packages/web/components/complex/all-pools-table-v2.tsx

```diff
@@ -184,31 +184,51 @@ export const AllPoolsTable: FunctionComponent<{
     [poolsPagesData]
   );
 
+  // If more than half of the pools have volume data, we should format the volume column.
+  // Otherwise, we should not format the volume column.
+  let volumePresenceCount = 0;
+  poolsData.forEach((pool) => {
+    if (pool.volume24hUsd) {
+      volumePresenceCount++;
+    }
+  });
+  let shouldFormatVolume = volumePresenceCount > poolsData.length / 2;
```

Comment:
This should be placed on a `useMemo`. Otherwise, it will call this expended calculation on every render

```suggestion
 const shouldFormatVolume = useMemo(() => {
    let volumePresenceCount = 0;
    poolsData.forEach((pool) => {
      if (pool.volume24hUsd) {
        volumePresenceCount++;
      }
    });
    return volumePresenceCount > poolsData.length / 2;
  }, [poolsData]);
```

Reply:
Also, you could return early if it reaches half of volume count.

Saving milliseconds.

But looks good